devicetypes/smartthings/zigbee-non-holdable-button.src/zigbee-non-holdable-button.groovy

```diff
@@ -0,0 +1,216 @@
+                              22: 70, 21: 50, 20: 50, 19: 30, 18: 30, 17: 15, 16: 1, 15: 0]
+            def minVolts = 15
+            def maxVolts = 28
+
+            if (volts < minVolts)
+                volts = minVolts
+            else if (volts > maxVolts)
+                volts = maxVolts
+            def pct = batteryMap[volts]
+            result.value = pct
+        } else {
+            def minVolts = 2.1
+            def maxVolts = 3.0
+            def pct = (volts - minVolts) / (maxVolts - minVolts)
+            def roundedPct = Math.round(pct * 100)
+            if (roundedPct <= 0)
+                roundedPct = 1
+            result.value = Math.min(100, roundedPct)
+        }
+
+    }
+
+    return result
+}
+
+private Map getBatteryPercentageResult(rawValue) {
+    log.debug "Battery Percentage rawValue = ${rawValue} -> ${rawValue / 2}%"
+    def result = [:]
+
+    if (0 <= rawValue && rawValue <= 200) {
+        result.name = 'battery'
+        result.translatable = true
+        result.descriptionText = "{{ device.displayName }} battery was {{ value }}%"
+        result.value = Math.round(rawValue / 2)
+    }
+
+    return result
+}
+
+private Map getButtonResult(value) {
+    def descriptionText
+    if (value == "pushed")
+        descriptionText = "${ device.displayName } was pushed"
+    else if (value == "held")
+        descriptionText = "${ device.displayName } was held"
+    else
+        descriptionText = "${ device.displayName } was pushed twice"
+    return [
+            name           : 'button',
+            value          : value,
+            descriptionText: descriptionText,
+            translatable   : true,
+            isStateChange  : true,
+            data           : [buttonNumber: 1]
+    ]
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+    zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS)
+}
+
+def refresh() {
+    log.debug "Refreshing Values"
+    def refreshCmds = []
+
+    refreshCmds += zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) +
+                   zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS) +
+                   zigbee.enrollResponse()
+    return refreshCmds
+}
+
+def configure() {
+    // Device-Watch allows 2 check-in misses from device + ping (plus 1 min lag time)
+    // enrolls with default periodic reporting until newer 5 min interval is confirmed
+    // Sets up low battery threshold reporting
+    sendEvent(name: "DeviceWatch-Enroll", displayed: false, value: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, scheme: "TRACKED", checkInterval: 6 * 60 * 60 + 1 * 60, lowBatteryThresholds: [15, 7, 3], offlinePingable: "1"])
+
+    log.debug "Configuring Reporting"
+    def configCmds = []
+    configCmds += zigbee.batteryConfig()
+    
+    return refresh() + configCmds + refresh() // send refresh cmds as part of config
```

Comment:
This will result in extra messages that aren't helpful. I suggest you instead return the following:

```
zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) +
zigbee.enrollResponse() +
zigbee.batteryConfig()
```

That will read the current battery level first which is useful since that's visible to the user, then do the IAS Zone enrollment since the user will likely next check that the button works, and finally setup the battery configuration. Unless your testing shows otherwise, I don't think there is any need to read the IAS Zone status attribute at all.